src/components/dashboard-settings/sync-apps/sync-apps.directive.js.coffee

```diff
@@ -0,0 +1,91 @@
+module = angular.module('impac.components.dashboard-settings.sync-apps',[])
+
+module.directive('dashboardSettingSyncApps', ($templateCache, $log, $http, $filter, $modal, ImpacMainSvc, ImpacRoutes, ImpacWidgetsSvc, ImpacTheming, poller) ->
+  return {
+    restrict: 'A',
+    scope: {
+    },
+    link: (scope, element, attrs) ->
+      scope.syncingApps = false
+      scope.hasConnectors = false
+
+      scope.showComponent = ImpacTheming.get().syncAppsConfig.show
+
+      openSyncAlertsModal =
+        open: null
+        engage: ()->
+          this.open = ()->
+            modalInstance = $modal.open({
+              animation: true
+              size: 'md'
+              templateUrl: 'alerts.tmpl.html'
+              appendTo: angular.element(document.getElementsByTagName('impac-dashboard'))
+              controller: ($scope, alerts) ->
+                $scope.alerts = alerts
+                $scope.ok = () ->
+                  modalInstance.close()
+              resolve:
+                alerts: ()->
+                  return scope.errors
+            })
+            this.open = null
+
+      refreshAllWidgets =
+        run: null
+        engage: ()->
+          this.run = ()->
+            ImpacWidgetsSvc.refreshAll()
+            this.run = null
+
+      ImpacMainSvc.load(true).then(
+        (config) ->
+          scope.orgUID = config.currentOrganization.uid
+
+          scope.syncingPoller = poller.get(ImpacRoutes.appInstancesSyncPath(scope.orgUID), {delay: 5000, smart: true})
```

Comment:
@xaun call has to be a singleton (= put in a service). Otherwise, a new poller is going to be registered everytime the directive is loaded (eg: everytime you switch companies).